bayesian/utility/db_gateway.py

```diff
@@ -63,3 +72,78 @@ def get_ca_data_from_graph(cls, ecosystem, package, version, vendor):
         elapsed_seconds = (datetime.now() - start).total_seconds()
         logger.info("Gremlin request took {} seconds.".format(elapsed_seconds))
         return query_result.json()
+
+
+class RdbAnalyses:
+    """RDB Interface class used by stack analyses flow.
+
+    Provides interfaces to save and read request post data for stack analyses v2.
+    """
+
+    def __init__(self, request_id, submit_time=None, manifest=None, deps=None):
+        """Set request id."""
+        self.request_id = request_id
+        self.submit_time = submit_time
+        self.manifest = manifest
+        self.deps = deps
+
+    def get_request_data(self):
+        """Read request data for given request id from RDS."""
+        db_result = fetch_sa_request(rdb, self.request_id)
+        if db_result is None:
+            error_message = 'Invalid request ID {}.'.format(self.request_id)
+            logger.exception(error_message)
+            raise RDBInvalidRequestException(error_message)
+        return db_result
+
+    def get_stack_result(self):
+        """Read and return stack result from RDS."""
+        return retrieve_worker_result(rdb, self.request_id, 'stack_aggregator_v2')
+
+    def get_recommendation_data(self):
+        """Read and return recommendation data from RDS."""
+        return retrieve_worker_result(rdb, self.request_id, 'recommendation_v2')
+
+    def save_post_request(self):
+        """Save the post request data into RDS."""
+        try:
+            insert_stmt = insert(StackAnalysisRequest).values(
+                id=self.request_id,
+                submitTime=self.submit_time,
+                requestJson={'manifest': self.manifest},
+                dep_snapshot=self.deps
+            )
+            do_update_stmt = insert_stmt.on_conflict_do_update(
+                index_elements=['id'],
+                set_=dict(dep_snapshot=self.deps)
+            )
+            rdb.session.execute(do_update_stmt)
+            rdb.session.commit()
+        except SQLAlchemyError as e:
+            logger.exception("Error updating log for request {}, exception {}".format(
+                self.request_id, e))
+            raise RDBSaveException('Error while saving request {}'.format(self.request_id))
```

Comment:
Use `raise YourCustomException(...) from e` pattern to preserve actual exception call trace.